[bug] Saving a new Research Article Update as Draft triggers the webhook

**To Reproduce**
Steps to reproduce the behaviour:

1. Start emulator backend.
2. Start frontend for emulator.
3. Login as precious plastic.
4. Edit the existing Research Article.
5. Create a new Update.
6. Create a title, description, and add a photo.
7. Press save as draft.
8. Check in the command line that the webhook was triggered.

Link to  emulator: https://github.com/ONEARMY/community-platform/blob/master/packages/documentation/docs/Backend%20Development/firebase-emulator.md#commands

**Expected behaviour**
The webhook should not be triggered.

Also, as Ben has pointed out in the comments, that if the article is a draft, it is currently also possible to trigger the webhook, which is bad mm'kay.

Hello @pizzaisdavid, thanks for the video explanation! 
Unfortunately I didn't manage to run correctly the firebase functions as the backend emulator throws a bunch of errors

I've synched my fork to the latest master branch commit and run the following commands

```
# on terminal 1
yarn install
yarn run frontend:for-emulated-backend:watch 
```

```
# on terminal 2
yarn run backend:emulator:watch
```

However the backend raises several error, some of which are `TS2307: Cannot find module 'oa-shared' or its corresponding type declarations` which I guess is the reason why my local environment is not running the firebase functions
I leave attached the backend logs in case someone knows how to fix them

[backend_logs.txt](https://github.com/user-attachments/files/16824048/backend_logs.txt)

On a more positive note, I've created a PR with to fix that should fix the issue but couldn't test it due to the mentioned errors

--- a/functions/src/Integrations/firebase-discord.test.ts
+++ b/functions/src/Integrations/firebase-discord.test.ts
@@ -1,3 +1,4 @@
+import { ResearchUpdateStatus } from 'oa-shared'
 import { handleResearchUpdatePublished } from './firebase-discord'
 
 import type { SimpleResearchArticle } from './firebase-discord'
@@ -54,6 +55,39 @@ describe('handle research article update change', () => {
           _id: 'bobmore',
           title: 'test',
           collaborators: ['Bob'],
+        },
+      ],
+    }
+
+    let wasMockSendMessagedCalled = false
+    const mockSendMessage = (_: string): Promise<any> => {
+      wasMockSendMessagedCalled = true
+      return Promise.resolve()
+    }
+
+    handleResearchUpdatePublished(
+      webhookUrl,
+      previousContent,
+      newContent,
+      mockSendMessage,
+    )
+    expect(wasMockSendMessagedCalled).toEqual(false)
+  })
+
+  it('should not send message when update is a draft', () => {
+    const webhookUrl = 'exmaple.com'
+    const previousContent: SimpleResearchArticle = {
+      slug: 'test',
+      updates: [],
+    }
+    const newContent: SimpleResearchArticle = {
+      slug: 'test',
+      updates: [
+        {
+          _id: 'bobmore',
+          title: 'test',
+          collaborators: ['Bob'],
+          status: ResearchUpdateStatus.DRAFT,
         },
       ],
     }

--- a/functions/src/Integrations/firebase-discord.ts
+++ b/functions/src/Integrations/firebase-discord.ts
@@ -1,6 +1,6 @@
 import axios from 'axios'
 import * as functions from 'firebase-functions'
-import { IModerationStatus } from 'oa-shared'
+import { IModerationStatus, ResearchUpdateStatus } from 'oa-shared'
 
 import { CONFIG } from '../config/config'
 
@@ -78,6 +78,7 @@ interface SimpleResearchArticleUpdate {
   _id: string
   title: string
   collaborators?: string[]
+  status?: ResearchUpdateStatus
 }
 
 export async function handleResearchUpdatePublished(
@@ -98,6 +99,11 @@ export async function handleResearchUpdatePublished(
 
   const newUpdateIndex = updatedContent.updates.length - 1
   const newUpdate = updatedContent.updates[newUpdateIndex]
+
+  if (newUpdate.status === ResearchUpdateStatus.DRAFT) {
+    console.log('Update is a draft')
+    return
+  }
 
   // On Research Updates, we actually expect the collaborators to be a single person
   // but it is a list.

